README.md

```diff
@@ -1 +1,206 @@
+
+
+
+
+
+### OAuth2 profiel: authorization_code
+
+![](img/authorization_code.png)
+
+Als uw applicatie met aprofielen of medewerkers wil laten inloggen en gegevens over deze gebruiker wenst op te halen moet de applicatie een geldig contract hebben met de "aprofiel" of "mprofiel" API.
+
+Technische implementatie details en een werkende NodeJS demo applicatie kan je hier vinden:
+https://github.com/digipolisantwerp/demo-oauth2-consumer_app_nodejs
+
+#### 1) de gebruiker naar de OAuth2 authorize applicatie redirecten
+
+Een authorization_code kan je bekomen door de gebruiker naar de OAuth2 Authorize applicatie te redirecten.
+
+
+#### 2) access_token bekomen
+
+Nadat de gebruiker is ingelogd en zijn goedkeuring(consent) heeft gegeven zal de gebruiker naar de redirect_uri geredirect worden die je hebt ingegeven in de details van uw applicatie op de API store;
+
+Deze redirect_uri bevat een *code* querystring parameter als de authenticatie succesvol is. Bijvoorbeeld:
+https://domain.com/callback?code=123
+
+*Request:*
+
+curl -X POST -d "client_id={{CLIENT_ID}}&client_secret={{CLIENT_SECRET}}&code={{CODE_FROM_PREV_STEP}}&grant_type=authorization_code" https://api-gw-p.antwerpen.be/org/service/v1/oauth2/token
+
+*Response:*
+{"refresh_token":"66f0c43c27a94ad4aa2d7574cf7b4465","token_type":"bearer","access_token":"a2824fb10b2a44b2b6f1a4aba382630a","expires_in":7200}
+
+#### 3) API oproepen
+curl -v -H "Authorization: Bearer {{ACCESS_TOKEN_FROM_PREVIOUS_STEP}}" 'https://api-gw-p.antwerpen.be/org/service/v1/resource'
+
+
+### OAuth2 profiel: implicit
+
+Hoewel het "implicit" OAuth2 profiel beschikbaar is op de API gateway raden we aan om deze niet te gebruiken.
+
+Voornamelijk omwille van onderstaande redenen:
++ Access_tokens die zijn aangemaakt met de implicit/token flow zijn slechts 2 uur gelding en kunnen niet verlengd worden.
++ Aangezien de client_secret niet gebruikt wordt om een access_token aan te vragen is dit OAuth2 profiel vatbaarder voor security issues.
+
+## Uitloggen
+
+## Het uitloggen van een gebruiker initiëren vanuit uw applicatie
+
+De OAuth2 authorizatie server bevat ook de mogelijkheid om gebruikers centraal te laten uitloggen.
+
+Om dit te bereiken moet je de gebruiker redirecten naar de "/logout/redirect/encrypted" endpoint van de OAuth2 authorizatie server met onderstaande querystring parameters.
+
+| Parameter | Verplicht | Omschrijving |
+| :---         |     :---:      |  :---   |
+| client_id     | true       | De client_id die je kan terugvinden bij uw applicatie in de API store.
+| service     | true       | De service waarmee de gebruiker zich geauthenticeerd heeft. Dit kan 'astad.aprofiel.v1' of 'astad.mprofiel.v1' zijn.      |
+| data     | true       | Een object met extra data ivm het uitloggen. Deze data is geencrypteerd volgens sha1 op basis van de client_secret.      |
```

Comment:
ge**ë**ncrypteerd volgens **SHA1**

Reply:
Is SHA1 voldoende veilig ?

Reply:
De opmerking ivm sha1 is mee op genomen in de backlog.